app/main.jsx

```diff
@@ -22,12 +24,16 @@ const ExampleApp = connect(
     </div>
 )
 
+const onProductsEnter = () => {
+  store.dispatch(loadProducts())
+}
+
 render(
   <Provider store={store}>
     <Router history={browserHistory}>
-      <Route path="/" component={Products}>
+      <Route path="/" component={Products} onEnter={onProductsEnter}>
```

Comment:
Because you have an index redirect to Products it's redundant to have Products load for two different routes. Also, as with the other pull request, I think that your main "/" route should be the App container that is being worked on right now.